src/modules/lifecycle_management.json

```diff
@@ -51,6 +51,133 @@
                 }
             ]
         },
+        {
+            "name": "session",
+            "summary": "Used by a Lifecycle provider to tell Firebolt that an app session has been started or updated. Firebolt can use this information to have context about the app when it makes API requests",
+            "params": [
+                {
+                    "name": "session",
+                    "required": true,
+                    "schema": {
+                        "$ref": "#/components/schemas/AppSession"
+                    }
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "const": null
+                }
+            },
+            "examples": [
+                {
+                    "name": "Starts or updates an app session",
+                    "params": [
+                        {
+                            "name": "session",
+                            "value": {
+                                "app": {
+                                    "id": "SomeApp",
+                                    "url": "https://some-app.firebolt.rdkcentral.com"
+                                },
+                                "runtime": {
+                                    "id": "WebBrowser-1"
+                                },
+                                "launch": {
+                                    "intent": {
+                                        "action": "search",
+                                        "data": {
+                                            "query": "Dog videos"
+                                        },
+                                        "context": {
+                                            "source": "voice"
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    ],
+                    "result": {
+                        "name": "defaultResult",
+                        "value": null
+                    }
+                }
+            ]
+        },
+        {
+            "name": "onRequestLaunch",
+            "summary": "Tells the provider that an app should be launched",
+            "params": [
+                {
+                    "name": "listen",
+                    "summary": "Whether or not to listen for requests to this interface method.",
+                    "schema": {
+                        "type": "boolean"
+                    }
+                }
+            ],
+            "tags": [
+                {
+                    "name": "event",
+                    "x-response": {
+                        "$ref": "#/components/schemas/LifecycleLaunchResponse",
+                        "examples": [
+                            {
+                                "success": true
```

Comment:
It'd be nice to have an auto-generated `launchError` RPC method that would be used to denote failure. 

If we do that, we can drop `success: true` here, and just return a `const: null`.

Question: should we just return the session here instead, or is there a reason to separate that into it's own API?

Reply:
Yeah if we have the error method, we should maybe use that.

We need the session API because this will only be used when Firebolt is the source of the launch (ie. an app calling Discovery.launch). App launches might happen in other paths and not through Firebolt.

Reply:
got it, makes sense